api/ngrok/v1alpha1/cloudendpoint_types.go

```diff
@@ -30,28 +30,45 @@ import (
 
 // CloudEndpointSpec defines the desired state of CloudEndpoint
 type CloudEndpointSpec struct {
-	// The unique URL for this cloud endpoint. This URL is the public address
+	// The unique URL for this cloud endpoint. This URL is the public address. The following formats are accepted
+	// Domain - example.org
+	//     When using the domain format you are only defining the domain. The scheme and port will be inferred.
+	// Origin - https://example.ngrok.app or https://example.ngrok.app:443 or tcp://1.tcp.ngrok.io:12345 or tls://example.ngrok.app
+	//     When using the origin format you are defining the protocol, domain and port. HTTP endpoints accept ports 80 or 443 with respective protocol.
+	// Scheme (shorthand) - https:// or tcp:// or tls:// or http://
+	//     When using scheme you are defining the protocol and will receive back a randomly assigned ngrok address.
+	// Empty - ``
+	//     When empty your endpoint will default to be https and receive back a randomly assigned ngrok address.
+	// Internal - some.domain.internal
+	//     When ending your url with .internal, an internal endpoint will be created. nternal Endpoints cannot be accessed directly, but rather
+	//     can only be accessed using the forward-internal traffic policy action.
+	//
 	// +kubebuilder:validation:Required
 	URL string `json:"url"`
 
-	// TrafficPolicyRef is a reference to the TrafficPolicy resource to attach to the Cloud Endpoint
+	// Reference to the TrafficPolicy resource to attach to the Cloud Endpoint
+	//
 	// +kubebuilder:validation:Optional
 	TrafficPolicyName string `json:"trafficPolicyName,omitempty"`
 
-	// TrafficPolicy allows inline definition of a TrafficPolicy object
+	// Allows inline definition of a TrafficPolicy object
+	//
 	// +kubebuilder:validation:Optional
 	TrafficPolicy *NgrokTrafficPolicySpec `json:"trafficPolicy,omitempty"`
 
-	// Description is a human-readable description of this cloud endpoint
+	// Human-readable description of this cloud endpoint
+	//
 	// +kubebuilder:default:=`Created by the ngrok-operator`
 	Description string `json:"description,omitempty"`
 
-	// Metadata is a string of arbitrary data associated with the object in the ngrok API/Dashboard
```

Comment:
Just a heads up, you might see these doc string formats with the struct field as the first word, mostly because we would just copy them from the ngrok-api-go client docstrings and then just make the properties camelcase instead of snake case.